src/notebook.jl

```diff
@@ -927,14 +928,48 @@ end
 # ╔═╡ dd495e00-d74d-47d4-a5d5-422fb147ec3b
 remove_linenums(x) = x
 
+# ╔═╡ e414c28a-8111-4821-ab25-21aff8289d26
+function remove_singleline_blocks(e::Expr)
+	if Meta.isexpr(e, :quote) || Meta.isexpr(e, :macrocall)
+		e
+	elseif Meta.isexpr(e, :block, 1)
+		remove_singleline_blocks(e.args[1])
+	else
+		Expr(
+			e.head, 
+			(remove_singleline_blocks(a) for a in e.args)...
+		)
+	end
+end
+
+# ╔═╡ c13e0f00-d3c4-4f1d-9531-84ed480c81f3
+remove_singleline_blocks(x) = x
+
+# ╔═╡ b765dbfe-4e58-4bb9-b1d6-aa4378d4e9c9
+expr_to_str(e; mod=@__MODULE__(), context::IO=devnull) = let
+	Computed;
+	
+	printed = sprint() do io
+		
+		Base.print(
+			IOContext(IOContext(io, :module => mod), context), 
+			remove_singleline_blocks(escape_syntax_to_esc_call(move_escape_calls_up(remove_linenums(e))))
+		)
+	end
+	replace(printed, r"#= line 0 =# ?" => "")
+end
+
+# ╔═╡ 227129bc-4415-4240-ad55-815bde65a5a1
+function Base.showerror(io::IO, error::CantStepifyThisYetException)
+	print(io, "CantStepifyThisYetException: Can't make `$(expr_to_str(error.expr))` into separate steps yet")
+end
+
 # ╔═╡ ab0a19b8-cf7c-4c4f-802a-f85eef81fc02
 function Base.show(io::IO, m::MIME"text/html", sd::SlottedDisplay)
 
 	d, e = sd.d, sd.e
 	
-	s = sprint() do iobuffer
-		print(IOContext(iobuffer, io), e |> remove_linenums)
-	end
```

Comment:
This PR also re-uses the `expr_to_str` function for SlottedDisplay, I'm guessing that we want this? @dralletje